src/herder/HerderPersistenceImpl.cpp

```diff
@@ -92,6 +93,50 @@ HerderPersistenceImpl::saveSCPHistory(uint32_t seq,
         }
     }
 
+    // save quorum information
+    for (auto const& p : qmap)
+    {
+        auto const& nodeID = p.first;
+        if (!p.second)
+        {
+            // skip node if we don't have its quorum set
+            continue;
+        }
+        auto qSetH = sha256(xdr::xdr_to_opaque(*p.second));
+        usedQSets.insert(std::make_pair(qSetH, p.second));
+
+        std::string nodeIDStrKey = KeyUtils::toStrKey(nodeID);
+        std::string qSetHHex(binToHex(qSetH));
+
+        auto prep = db.getPreparedStatement(
+            "UPDATE quoruminfo SET qsethash = :h WHERE nodeid = :id");
```

Comment:
We're on a version of sql that supports upserts (INSERT ... ON CONFLICT SET ...) now, if you want to simplify this a bit.

Reply:
I am not too worried about this one yet, and sqlite doesn't support it